statsmodels/discrete/tests/test_discrete.py

```diff
@@ -100,13 +107,39 @@ def test_bic(self):
         assert_almost_equal(self.res1.bic, self.res2.bic, DECIMAL_3)
 
     def test_predict(self):
-        assert_almost_equal(self.res1.model.predict(self.res1.params),
-                            self.res2.phat, DECIMAL_4)
+
+        fitted = self.res1.model.predict(self.res1.params)
+
+        res2_fitted = getattr(self.res2, 'phat', None)
+        if res2_fitted is None:
+            res2_fitted = np.exp(getattr(self.res2, 'linpred', None))
+        if res2_fitted is not None:
+            assert_allclose(fitted[:len(res2_fitted)], res2_fitted, rtol=5e-4)
+
+        # fittedvalues in discrete are currently linear prediction
+        assert_allclose(self.res1.fittedvalues[:len(res2_fitted)],
+                        res2_fitted, atol=5e-4, rtol=5e-4)
+
+        if self.res1.params.ndim == 2:
```

Comment:
From here down seems like it should be a separate test

Reply:
Maybe test_resid_response?  Could then remove a smoke test in CheckBinaryResults